admin-dev/themes/new-theme/js/pages/product/edit/combinations-manager.js

```diff
@@ -27,37 +27,101 @@ import ProductMap from '@pages/product/product-map';
+    this.$combinationsContainer = $(ProductMap.combinations.combinationsContainer);
+    this.combinationIdInputsSelector = ProductMap.combinations.combinationIdInputsSelector;
     this.initialized = false;
+    this.combinationsService = new CombinationsService(this.getProductId());
+
     this.init();
 
     return {};
   }
 
   init() {
-    const productId = this.getProductId();
     this.paginator = new DynamicPaginator(
       ProductMap.combinations.paginationContainer,
-      new CombinationsService(productId),
+      this.combinationsService,
       new CombinationsGridRenderer(),
     );
-
+    this.initSubmittableInputs();
+    this.onDefaultCombinationSelection();
     // Paginate to first page when tab is shown
     this.$productForm.find(ProductMap.combinations.navigationTab).on('shown.bs.tab', () => this.firstInit());
   }
 
+  initSubmittableInputs() {
+    const combinationToken = this.getCombinationToken();
+
+    new SubmittableInput('.combination-quantity', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[quantity][value]': input.value, 'combination_item[_token]': combinationToken},
+      );
+    });
+
+    new SubmittableInput('.combination-impact-on-price', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[impact_on_price][value]': input.value, 'combination_item[_token]': combinationToken},
+      );
+    });
+
+    new SubmittableInput('.combination-reference', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[reference][value]': input.value, 'combination_item[_token]': combinationToken},
+      );
+    });
+  }
+
+  onDefaultCombinationSelection() {
+    this.$combinationsContainer.on('change', ProductMap.combinations.isDefaultInputsSelector, async (e) => {
+      const checkedInputs = this.$combinationsContainer.find(
+        `${ProductMap.combinations.isDefaultInputsSelector}:checked`,
+      );
+
+      // stop loop when only one input is checked
+      if (checkedInputs.length === 1) {
+        return;
+      }
+
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(e.currentTarget),
+        {
+          'combination_item[is_default]': e.currentTarget.value,
+          'combination_item[_token]': this.getCombinationToken(),
+        },
+      );
+
+      $.each(checkedInputs, (index, input) => {
+        if (this.findCombinationId(input) !== this.findCombinationId(e.currentTarget)) {
+          $(input).prop('checked', false);
+        }
+      });
+    });
+  }
+
   firstInit() {
     if (this.initialized) {
       return;
     }
 
     this.initialized = true;
-    this.paginator.paginate(1);
+    this.paginator.paginateToLastViewed();
```

Comment:
it seems quite useful to me. If you switch product tabs (like to check something in another tab) you will get back on same page where you left of.

Reply:
But that's already what's supposed to happen right now, thanks to the `firstInit` check It only forces the automatic pagination when you enable the tab the first time

Then if you go the another tab and come back the presentation simply doesn't change because `this.initialized` is already true, so no automatic pagination is required, and re-paginating is also not necessary because it would add a delay

Reply:
I still don't understand the interest of this, I re-tested the current develop branch and the component is indeed initialized only once so it's already enough